app/src/main/java/com/swent/suddenbump/model/user/UserRepositoryFirestore.kt

```diff
@@ -1100,18 +1097,15 @@ class UserRepositoryFirestore(private val db: FirebaseFirestore, private val con
    * @return `true` if the user is logged in, `false` otherwise.
    */
   override fun isUserLoggedIn(): Boolean {
-    return sharedPreferences.getBoolean("isLoggedIn", false)
+    return sharedPreferencesManager.getBoolean("isLoggedIn")
   }
 
   /**
    * Logs out the current user by updating shared preferences to remove login status and user ID.
    */
   override fun logoutUser() {
-    with(sharedPreferences.edit()) {
-      putBoolean("isLoggedIn", false)
-      putString("userId", null)
-      apply()
-    }
+    workerScheduler.unscheduleLocationUpdateWorker()
+    sharedPreferencesManager.clearPreferences()
```

Comment:
Very good implementation here: it's important to both unschedule the LocationUpdateWorker and clear the user preferences upon logging out.